dotnet/src/Extism.Sdk/Context.cs

```diff
@@ -0,0 +1,156 @@
+using System.Runtime.InteropServices;
+
+namespace Extism.Sdk.Native;
+
+/// <summary>
+/// Represents an Extism context through which you can load <see cref="Plugin"/>s.
+/// </summary>
+public class Context : IDisposable
+{
+    private bool _disposed;
+
+    /// <summary>
+    /// Initialize a new Extism Context.
+    /// </summary>
+    public Context()
+    {
+        NativeHandle = LibExtism.extism_context_new();
+    }
+
+    /// <summary>
+    /// Native pointer to the Extism Context.
+    /// </summary>
+    public IntPtr NativeHandle { get; }
+
+    /// <summary>
+    /// Loads an Extism <see cref="Plugin"/>.
+    /// </summary>
+    /// <param name="wasm">The plugin WASM bytes.</param>
+    /// <param name="withWasi">Enable/Disable WASI.</param>
+    /// <returns></returns>
+    public Plugin CreatePlugin(ReadOnlySpan<byte> wasm, bool withWasi)
```

Comment:
I think this is good for now, we'll want to eventually support passing in raw bytes or a manifest object: https://extism.org/docs/concepts/manifest/

extism_plugin_new takes a byte array that can be either the raw bytes of the module or a json serialized Manifest.

Reply:
I think I will just update the docs that both are supported, in another PR we can add another layer `Extism.Sdk.Json` that adds full Json Serialization/Deserialization support and makes the config a proper object too